eip-0045.md

```diff
@@ -0,0 +1,114 @@
+
+* value - collected Ergs in this contract
+* length - across which we target to redistribute the value
+* ratio - how much of collected fees does this contract receive
+
+and such contract would pay to the miner reward equal to (value / length). We can have multiple such contracts with different values of parameters however the total sum of all ratios must be always equal to 1 (100%). A miner, who collects the storage rent fees, sends all fees to the Smart Contract implementing this logic. This Smart Contract would then redistribute the collected storage fees.
+
+A trait of such contract with small testing demo can be explored on the following Ergo playground [link](https://scastie.scala-lang.org/RHIwF2hwTjSJLKmRLiXKpA) but with misbehaving proper increasing blockchain height for created transaction the `heightIncreased` variable is not used in final `SigmaProp` as it evaluates always to false (height remains 0 for every created transaction and constructor for defining height of transaction is missing in Ergo playground).<br>
+Please note this is just for demonstration purposes.
+
+Code snippet of Smart Contract in ErgoScript:
+```scala
+    val contractOut = OUTPUTS(0)
+
+	// nft identifier is preserved
+    val correctNftOut = contractOut.tokens(0)._1 == redistributionNft
+
+	// created for current height
+	val heightCorrect = contractOut.creationInfo._1 == HEIGHT
+
+	// box spendable only once per block
+   	val heightIncreased = HEIGHT > SELF.creationInfo._1
+
+   	// script is preserved
+    val correctProposition = contractOut.propositionBytes == SELF.propositionBytes
+
+    // we can optionally lock miner's reward
+      
+    val outTxValue = OUTPUTS(1).value
+    val minerReward = outTxValue - contractOut.value
+
+   	val values = SELF.R4[Coll[Long]].get
+	val indices = values.indices
+
+	val newValues = contractOut.R4[Coll[Long]].get
+   
+	val claimableReward = indices.fold(0L, { (sum: Long, i: Int) =>  sum + values(i) / params(i)._1 })
+
+   	val correctMinerClaim = minerReward <= claimableReward
+     	
+    // this should always be positive
+    val collectedFees = contractOut.value - (SELF.value - minerReward)
+
+    val correctValues = allOf(Coll({
+      	indices.forall({ (i: Int) => ({
+            // optionally we can calculate from what miner actually claimed if he does not pay txFee
+            // (minerReward * params(i)._2) / 100
+            val partialClaim = (values(i) / params(i)._1)
+            // we calculate what will be added to contract
+            val addedValue = (collectedFees * params(i)._2) / 100
+            
+            newValues(i) >= values(i) - partialClaim + addedValue
+            })
+        })
+    }))
+      
+    sigmaProp(
+      	allOf(
+       		Coll(
+           	    correctNftOut,
+                heightCorrect,
+                heightIncreased,
+                correctProposition,
+                correctMinerClaim,
+                correctValues
+            )
+        )
+    )
+```
```

Comment:
@The-Huginn, please explictily describe the box which should be protected by this contract and if it has additional registers.